homeassistant/components/rainmachine/__init__.py

```diff
@@ -158,6 +182,39 @@ async def async_update(api_category: str) -> dict:
 
     entry.async_on_unload(entry.add_update_listener(async_reload_entry))
 
+    async def async_pause_watering(call: ServiceCall) -> None:
+        """Pause watering for a set number of seconds."""
+        controllers = async_get_controllers_for_service_call(hass, call)
+        tasks = [
+            controller.watering.pause_all(call.data[CONF_SECONDS])
+            for controller in controllers
+        ]
+        await asyncio.gather(*tasks)
+        await async_update_programs_and_zones(hass, entry)
+
+    async def async_stop_all(call: ServiceCall) -> None:
+        """Stop all watering."""
+        controllers = async_get_controllers_for_service_call(hass, call)
+        tasks = [controller.watering.stop_all() for controller in controllers]
+        await asyncio.gather(*tasks)
+        await async_update_programs_and_zones(hass, entry)
+
+    async def async_unpause_watering(call: ServiceCall) -> None:
+        """Unpause watering."""
+        controllers = async_get_controllers_for_service_call(hass, call)
+        tasks = [controller.watering.unpause_all() for controller in controllers]
+        await asyncio.gather(*tasks)
+        await async_update_programs_and_zones(hass, entry)
+
+    for service_name, method in (
+        (SERVICE_PAUSE_WATERING, async_pause_watering),
+        (SERVICE_STOP_ALL, async_stop_all),
+        (SERVICE_UNPAUSE_WATERING, async_unpause_watering),
+    ):
+        if hass.services.has_service(DOMAIN, service_name):
+            continue
+        hass.services.async_register(DOMAIN, service_name, method)
```

Comment:
Can you tell me more? These services don't require a schema, per se...unless I need to explicitly check for a device ID?

Reply:
Yeah, we need to validate the key for the device id.

Reply:
Does this need to be expanded to other triggers, like an area ID?￼

Reply:
That's up to you. With the device selector, only device ids will be valid in the frontend. I'd maybe start with just device id. If you want to expand on that, it can be added later.